packages/@lwc/engine-core/src/framework/renderer.ts

```diff
@@ -24,7 +24,7 @@ export interface RendererAPI {
     nextSibling: (node: N) => N | null;
     attachShadow: (element: E, options: ShadowRootInit) => N;
     getProperty: (node: N, key: string) => any;
-    setProperty: (node: N, key: string, value: any) => void;
+    setProperty: (node: N, key: string, value: any, attributeName: string) => void;
```

Comment:
I don't understand why the new argument is now required?

Reply:
There is validation in `setProperty` implementation that verifies if the property is supported by the element. The error message needs the `htmlPropertyToAttribute`(implemented in `@lwc/shared`) conversion. 
Instead of `@lwc/engine-dom` renderer relying on this function from `@lwc/shared`, the engine-core will get value and pass it to `setProperty`.

Reply:
let's chat about it... there has to be another way.